source/interface/azure_iot_mqtt.h

```diff
@@ -0,0 +1,380 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+/**
+ * @file azure_iot_mqtt.h
+ *
+ * @brief Definition for the Azure IoT MQTT.
+ *
+ */
+
+#ifndef AZURE_IOT_MQTT_H
+#define AZURE_IOT_MQTT_H
+
+#include <stdint.h>
+#include <stddef.h>
+
+#if defined( __cplusplus ) || ( defined( __STDC_VERSION__ ) && ( __STDC_VERSION__ >= 199901L ) )
+    #include <stdbool.h>
+#elif !defined( bool ) && !defined( false ) && !defined( true )
+    #define bool     int8_t
+    #define false    ( int8_t ) 0
+    #define true     ( int8_t ) 1
+#endif
+
+#include "azure_iot_mqtt_port.h"
+#include "azure_iot_transport_interface.h"
+
+#define azureiotmqttPACKET_TYPE_CONNECT        ( ( uint8_t ) 0x10U )  /** @brief CONNECT (client-to-server). */
+#define azureiotmqttPACKET_TYPE_CONNACK        ( ( uint8_t ) 0x20U )  /** @brief CONNACK (server-to-client). */
+#define azureiotmqttPACKET_TYPE_PUBLISH        ( ( uint8_t ) 0x30U )  /** @brief PUBLISH (bidirectional). */
+#define azureiotmqttPACKET_TYPE_PUBACK         ( ( uint8_t ) 0x40U )  /** @brief PUBACK (bidirectional). */
+#define azureiotmqttPACKET_TYPE_PUBREC         ( ( uint8_t ) 0x50U )  /** @brief PUBREC (bidirectional). */
+#define azureiotmqttPACKET_TYPE_PUBREL         ( ( uint8_t ) 0x62U )  /** @brief PUBREL (bidirectional). */
+#define azureiotmqttPACKET_TYPE_PUBCOMP        ( ( uint8_t ) 0x70U )  /** @brief PUBCOMP (bidirectional). */
+#define azureiotmqttPACKET_TYPE_SUBSCRIBE      ( ( uint8_t ) 0x82U )  /** @brief SUBSCRIBE (client-to-server). */
+#define azureiotmqttPACKET_TYPE_SUBACK         ( ( uint8_t ) 0x90U )  /** @brief SUBACK (server-to-client). */
+#define azureiotmqttPACKET_TYPE_UNSUBSCRIBE    ( ( uint8_t ) 0xA2U )  /** @brief UNSUBSCRIBE (client-to-server). */
+#define azureiotmqttPACKET_TYPE_UNSUBACK       ( ( uint8_t ) 0xB0U )  /** @brief UNSUBACK (server-to-client). */
+#define azureiotmqttPACKET_TYPE_PINGREQ        ( ( uint8_t ) 0xC0U )  /** @brief PINGREQ (client-to-server). */
+#define azureiotmqttPACKET_TYPE_PINGRESP       ( ( uint8_t ) 0xD0U )  /** @brief PINGRESP (server-to-client). */
+#define azureiotmqttPACKET_TYPE_DISCONNECT     ( ( uint8_t ) 0xE0U )  /** @brief DISCONNECT (client-to-server). */
+
+
+typedef enum AzureIoTMQTTQoS
+{
+    eAzureIoTMQTTQoS0 = 0, /** Delivery at most once. */
```

Comment:
should these be more descriptive? eAzureIoTMQTTQoS0 -> eAzureIoTMQTTDeliveryAtMostOnce

Reply:
enum type is AzureIoTMQTTQoS so It would be correct to name them based on MQTT Quality of service

---
In reply to: [598023132](https://github.com/Azure/azure-iot-middleware-freertos/pull/44#discussion_r598023132) [](ancestors = 598023132)